src/services/drpcEvents/index.ts

```diff
@@ -146,4 +153,85 @@ export default class DrpcEvents {
       })
     }
   }
+
+  private async handleSubmitQueryResponse(request: DrpcRequest, agentConnectionId: string) {
+    let queryId: string | null = null
+    try {
+      this.logger.info(
+        'DRPC response (%s) received on connection %s of method %s',
+        request.id,
+        agentConnectionId,
+        request.method
+      )
+
+      let params: SubmitQueryRPCParams
+      try {
+        params = submitQueryRpcParams.parse(request.params)
```

Comment:
you shouldn't reuse the same parser here. You should only need the `queryId` and the `emissions` I think.

Reply:
Yeah I wasn't sure if we'd want to also keep the original query just in case there is a mix up or sth so we can prove that whoever is answering is providing an answer only to that one query...but if you think it's unnecessary then I'm happy to remove

Reply:
yep please change to using a dedicated parser

Reply:
It looks like this has been updated 

```js
      let params: SubmitQueryResponseRPCParams
      try {
        params = submitQueryResponseRpcParams.parse(request.params)
```